eclair-core/src/test/scala/fr/acinq/eclair/io/SwitchboardSpec.scala

```diff
@@ -141,6 +144,66 @@ class SwitchboardSpec extends TestKitBaseClass with AnyFunSuiteLike {
     peer.expectMsg(Peer.GetPeerInfo(Some(probe.ref.toTyped)))
   }
 
+  test("track nodes with incoming connections that do not have a channel") {
+    val nodeParams = Alice.nodeParams.copy(peerConnectionConf = Alice.nodeParams.peerConnectionConf.copy(maxNoChannels = 2))
+    val (probe, peer, peerConnection, channel) = (TestProbe(), TestProbe(), TestProbe(), TestProbe())
+    val hasChannelsNodeId1 = randomKey().publicKey
+    val hasChannelsNodeId2 = randomKey().publicKey
+    val unknownNodeId1 = randomKey().publicKey
+    val unknownNodeId2 = randomKey().publicKey
+    val switchboard = TestActorRef(new Switchboard(nodeParams, FakePeerFactory(probe, peer)))
+    switchboard ! Switchboard.Init(Nil)
+
+    // Do not track nodes we connect to.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, randomKey().publicKey, outgoing = true)
+    peer.expectMsgType[Peer.Init]
+
+    // Do not track an incoming connection from a peer we have a channel with.
+    switchboard ! ChannelIdAssigned(channel.ref, hasChannelsNodeId1, randomBytes32(), randomBytes32())
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, hasChannelsNodeId1, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+
+    // We do not yet have channels with these peers, so we track their incoming connections.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, unknownNodeId1, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, unknownNodeId2, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+
+    // Disconnect the oldest tracked peer when an incoming connection from a peer without channels connects.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, randomKey().publicKey, outgoing = false)
+    peer.fishForMessage() {
+      case d: Peer.Disconnect => d.nodeId == unknownNodeId1
+      case _: Peer.Init => false
+    }
+
+    // Do not disconnect an old peer when a peer with channels connects.
+    switchboard ! ChannelIdAssigned(channel.ref, hasChannelsNodeId2, randomBytes32(), randomBytes32())
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, hasChannelsNodeId2, outgoing = false)
+    peer.expectMsgType[Peer.Init]
+    peer.expectNoMessage(100 millis)
+
+    // Disconnect the next oldest tracked peer when an incoming connection from a peer without channels connects.
+    switchboard ! PeerConnection.Authenticated(peerConnection.ref, randomKey().publicKey, outgoing = false)
+    peer.fishForMessage()  {
```

Comment:
nit:
```suggestion
    peer.fishForMessage() {
```

Reply:
fixed in [a00f448](https://github.com/ACINQ/eclair/pull/2601/commits/a00f448df3d45b3ab26952948a3c47f6cfeec9c0)